lib/rubocop/directive_comment.rb

```diff
@@ -21,18 +23,11 @@ def self.before_comment(line)
       line.split(DIRECTIVE_COMMENT_REGEXP).first
     end
 
-    attr_reader :comment
+    attr_reader :comment, :switch, :cops_string
```

Comment:
I think we need a more descriptive name than `switch` because I'm not even sure what it means here. Maybe `mode`?

Also since you removed `cops` I think `@cops` would be more straightforward than `@cops_string`.

Reply:
Thanks, replaced